### PR TITLE
docs: proposal for new repo koordetector

### PR DESCRIPTION
**Pull request by songtao98**
Signed-off-by: songtao98 <songtao2603060@gmail.com>

### Ⅰ. Describe what this PR does
Add proposals that clarifies the design for new repo koordetector.

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: docs/proposals/20221205-new-repo-koordetector.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+eBPF, extended Berkeley Packet Filter. See: https://ebpf.io/
+
+OpenAnolis: https://gitee.com/anolis
+
+Prometheus: https://prometheus.io/
+
+## Summary
+
+To support Koordinator's Interference Detection feature, some advanced metrics need to be collected by multiple technologies such as CPU schedule latency by eBPF. This proposal is intended to develop a new sub-repo for Koodinator named `Koordetector` which covers many underlying kernel metrics collection tools implemented with eBPF technology or others, and expands Koordinator's ability to collect metrics in the form of plug-ins.
```
Comment (zwzhang0107) [inline docs/proposals/20221205-new-repo-koordetector.md:47]: not only metrics collection, but also detect and analysis.

File: docs/proposals/20221205-new-repo-koordetector.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+#### Story 2
+
+As a cluster administrator, I want to use Koordinator with small cost and control of the metric collectors. 
+
+### Design
+
+![koordetector](../images/koordetector.jpg)
+
+The whole structure of `Koordetector` looks similar to `Koordlet`, but without `QosManager` related components. `StateInformer`, `MetricsCache` are also necessary for collectors to run.
```
Comment (zwzhang0107) [inline docs/proposals/20221205-new-repo-koordetector.md:84]: just introduce koordetector itself, no need to compare with others.
FYI:
`advisor manager`: manage a set of metrics collector, and also support external tools as prometheus format.
`tsdb`: cache data for analysis.
`analyser`: analysis metrics to detect interference.
`exporter`: expose data API for koordlet, and report to `koord metric server` in the future.

File: docs/proposals/20221205-new-repo-koordetector.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+- Define the concepts, responsibilities and basic structure of the new repo `Koordetector`
+
+### Non-Goals/Future Work
+
+- [Future Plan] Design metric reporting CRDs or APIs. 
+
+## Proposal
+
+### User Stories
```
Comment (zwzhang0107) [inline docs/proposals/20221205-new-repo-koordetector.md:70]: story 1: collect advanced metrics for interference detection.
story 2: adapt third-part tools for extension.

File: docs/proposals/20221205-new-repo-koordetector.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    Stop() error
+}
+```
+
+At the very beginning of metrics collection and interference detection, the metrics collected are reported by Prometheus as below. 
+
+```go
+Metric = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: "koordetector",
+  	Name:      "metric_name",
```
Comment (saintube) [inline docs/proposals/20221205-new-repo-koordetector.md:128]: indent problem

File: docs/proposals/20221205-new-repo-koordetector.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+type Collector interface {
+    Name() string
+    Run(stop <-chan struct{}) ()
+    Stop() error
+}
+
+type ExternalCollector interface {
+    Name() string
+  	URL() string
```
[the remark this answers is not in the document]
Comment (saintube) [reply docs/proposals/20221205-new-repo-koordetector.md:117]: it still exists. please check the raw content

```
type ExternalAdvisor interface {
    Name() string
  	URL() string
}
```

**Review by saintube — APPROVED** [review]
/lgtm

**Pull request merged**